test/agentchat/test_conversable_agent.py

```diff
@@ -402,6 +403,10 @@ def exec_sh(script: str) -> None:
         assert agent.function_map["sh"] == exec_sh
 
 
+@patch("autogen.oai.client.OpenAI", new=object)
+# Mocking an empty OpenAI client (simulating default OpenAIWrapper behaviour in case `openai` is not installed)
+# Without this fix the test will pass if no `openai` is installed and fail if it is
+# With the mocked `autogen.oai.client.OpenAI`` the test consistently passes no matter if `openai` is present or not
```

Comment:
@davorrunje is this clear to you? If openai is installed, why do we need to mock `autogen.oai.client.OpenAI`?
The first sentence says the client is used for simulating OpenAIWrapper when `openai` is not installed, and the second sentence says when `openai` is not installed, the test will pass without this fix. I'm confused.

Reply:
@sonichi Yes, it is clear to me. In test, we need to construct a ConversableAgent and if openai is installed, a key must be set in env var or the test would fail. If openai is not installed, then OpenAI variable in clint.py is set to object so everything works

However, I agree with you this is all confusing. A much cleaner way would be to refactor `ConversableAgent._wrap_function()` and make it static to avoid the need for calling OpenAI constructor in the first place.

@maxim-saplin I'll create a PR to this branch soon to fix this.

Reply:
@maxim-saplin can you please allow me the write access to the repo?

Reply:
Actually, this was my mistake. I had OPENAI_API_KEY env variable set locally, which caused tests to pass regardless of whether openai lib was being installed. It is fixed now and I am passing False instead of {} for llm_config in tests.